src/mixins/raster-layer-cross-section-terrain-mixin.js

```diff
@@ -0,0 +1,325 @@
+      chart,
+      table,
+      CrossSectionTerrainConfigDefinitionObject.key,
+      this,
+      layerName,
+      lastFilteredSize
+    )
+
+    _vega_property_output_state = materializePropDescriptors(
+      raster_layer_context,
+      prop_descriptors,
+      state
+    )
+    const {
+      sql_parser_transforms,
+      vega_data_formats,
+      vega_transforms,
+      vega_scales,
+      mark_properties
+    } = _vega_property_output_state.flatten()
+
+    assert(vega_data_formats.length <= 1)
+
+    if (typeof filter === "string" && filter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: filter
+      })
+    }
+
+    const post_filter_transform = create_post_filter_transform(
+      state.postFilters
+    )
+
+    if (post_filter_transform) {
+      sql_parser_transforms.push(post_filter_transform)
+    }
+
+    if (typeof globalFilter === "string" && globalFilter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: globalFilter
+      })
+    }
+
+    const data = [
+      {
+        name: layerName,
+        ...(vega_data_formats.length ? vega_data_formats[0] : {}),
+        sql: parser.writeSQL({
+          type: "root",
+          source: table,
+          transform: sql_parser_transforms
+        }),
+        enableHitTesting: state.enableHitTesting
+      },
+      ...vega_transforms
+    ]
+
+    const marks = [
+      {
+        type: CrossSectionTerrainConfigDefinitionObject.key,
+        from: {
+          data: layerName
+        },
+        properties: Object.assign({}, ...mark_properties)
+      }
+    ]
+
+    return { data, scales: vega_scales, marks }
+  }
+
+  function isCrossFilterObject(obj) {
+    if (
+      Object.hasOwn(obj, "type") &&
+      Object.hasOwn(obj, "getCrossfilter") &&
+      Object.hasOwn(obj, "filter")
+    ) {
+      // is this safe to assume that if these properties exist, then the object
+      // is a crossfilter-related object?
+      return true
+    }
+    return false
+  }
+
+  function createCrossfilterDimensionProxy(range) {
+    let range_ = [...range]
+    return {
+      type: "crossfilter_dimension_proxy",
+      getFilter: () => [range_],
+      filter: new_range => {
+        range_ = new_range
+      },
+      dispose: () => {}
+    }
+  }
+
+  _layer.xDim = createRasterLayerGetterSetter(_layer, null, new_val => {
+    if (isCrossFilterObject(new_val)) {
+      return new_val
+    }
+    return createCrossfilterDimensionProxy(new_val)
+  })
+  _layer.yDim = createRasterLayerGetterSetter(_layer, null, new_val => {
+    if (isCrossFilterObject(new_val)) {
+      return new_val
+    }
+    return createCrossfilterDimensionProxy(new_val)
+  })
+
+  let _vega = null
+  const _cf = null
+  _layer.crossfilter = createRasterLayerGetterSetter(_layer, _cf)
+
+  // eslint-disable-next-line no-unused-vars
+  _layer._genVega = function(chart, layerName, group, query) {
+    const realLayerName = layerName
```

Comment:
Can remove

Reply:
I'm unsure about this one.  What he did is very explicit, and makes me wonder if some modification happens to `layerName` when it is passed into the `__genVega` call that mutates the referenced var, thereby messing it up if you pass the same variable to `getFilterString()`.  I don't think it's worth it.

Reply:
Layer name is a string primitive so references won't be a thing here it will just receive the string value
<img width="320" alt="Screen Shot 2023-05-12 at 10 23 00 AM" src="https://github.com/heavyai/heavyai-charting/assets/1031659/8d7bf2fe-d161-4cbd-8633-79f2b63338ee">

and changing a parameter reference won't change the  reference of the variable passed in. 
<img width="297" alt="Screen Shot 2023-05-12 at 10 33 46 AM" src="https://github.com/heavyai/heavyai-charting/assets/1031659/11055d85-af5b-44aa-8a30-514149dfd48a">

If we're imagining that `layerName` is an object and properties of it may be mutated, then this re-assignment of the reference does nothing to prevent that.
<img width="412" alt="Screen Shot 2023-05-12 at 10 35 49 AM" src="https://github.com/heavyai/heavyai-charting/assets/1031659/4b19203d-2beb-4620-b159-b1c4622ec8db">

But if you want to choose fear over love than I understand 😉

Reply:
While I completely agree with you, it's just seems really strange that he would explicitly make that variable if there was no need.....honestly I don't see what could be wrong with just using the same variable (everything you said makes perfect sense), but for now at least, I'm sadly going to choose fear in lieu of the tiny chance of introducing some hard-to-detect bug in the future :)